costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -465,12 +465,14 @@ void Costmap2DROS::mapUpdateLoop(double frequency)
       publisher_->updateBounds(x0, xn, y0, yn);
 
       ros::Time now = ros::Time::now();
-      if (last_publish_ + publish_cycle < now)
+      ROS_WARN_COND(now < last_publish_, "ROS Time jumped backwards by %.3f s. Publishing costmaps anyway.", (last_publish_.toSec()-now.toSec()));
```

Comment:
S/F: shorten to `(last_publish_ - now).toSec()`; that will also remove imprecision introduced by subtracting two large double numbers from each other